src/azure-cli/azure/cli/command_modules/monitor/operations/monitor_clone_util.py

```diff
@@ -0,0 +1,107 @@
+    alert_rule.scopes = [target_resource]
+    resource_group_name, name = _parse_id(target_resource).values()
+    name = CLONED_NAME.format(name, _gen_guid())
+    return monitor_client.metric_alerts.create_or_update(resource_group_name=resource_group_name,
+                                                         rule_name=name,
+                                                         parameters=alert_rule)
+
+
+def _clone_monitor_metrics_alerts(cmd, source_resource, target_resource, always_clone=False):
+    same_rp, same_sub = _is_resource_type_same_and_sub_same(source_resource, target_resource)
+    if not same_rp:
+        raise CLIError('The target resource should be the same type with the source resource')
+    source_monitor_client = cf_monitor(cmd.cli_ctx, subscription_id=parse_resource_id(source_resource)['subscription'])
+    target_monitor_client = cf_monitor(cmd.cli_ctx, subscription_id=parse_resource_id(target_resource)['subscription'])
+    updated_metrics_alert_rules = []
+    action_group_mapping = {}
+    ErrorResponseException = cmd.get_models('ErrorResponseException', operation_group='metric_alerts')
+    for alert_rule in _gen_metrics_alert_rules_clone_list(source_monitor_client, source_resource, target_resource):
+        if always_clone or not same_sub:
+            alert_rule = _clone_and_replace_action_group(source_monitor_client,
+                                                         target_monitor_client,
+                                                         cmd,
+                                                         alert_rule,
+                                                         action_group_mapping,
+                                                         target_resource)
+            alert_rule = _clone_alert_rule(target_monitor_client,
+                                           alert_rule,
+                                           target_resource)
+        else:
+            try:
+                alert_rule = _add_into_existing_scopes(source_monitor_client,
+                                                       alert_rule,
+                                                       target_resource)
+            except ErrorResponseException as ex:  # Create new alert rule
+                if ex.response.status_code == 400:
+                    alert_rule = _clone_alert_rule(target_monitor_client,
+                                                   alert_rule,
+                                                   target_resource)
+                else:
+                    raise ex
+        updated_metrics_alert_rules.append(alert_rule)
+
+    return updated_metrics_alert_rules
+
+
+def _is_resource_type_same_and_sub_same(source_resource, target_resource):
+    source_dict = parse_resource_id(source_resource)
+    target_dict = parse_resource_id(target_resource)
+    same_rp = source_dict['namespace'] == target_dict['namespace'] and source_dict['type'] == target_dict['type']
```

Comment:
should be case-insensitive here?

Reply:
I mean if target_dict['namespace'] should be case-insensitive?

---
In reply to: [394248241](https://github.com/Azure/azure-cli/pull/12233#discussion_r394248241) [](ancestors = 394248241)

Reply:
yes. I have make source_resource and target_resource to be lower case. So everything in the dict should be lower case